### DISCUSSION TITLE
Temp `*.quarto_ipynb` files are not deleted after publishing

### DISCUSSION THREAD

**Discussion by cbrnr** [quarto-dev/quarto-cli#11480] [Q&A] [answered]
I have a weird issue that unfortunately I cannot reproduce with a minimal example, so I'll try to describe what's happening as best as I can.

I have a project-like folder structure with an empty `_quarto.yml` in the root. There are several subfolders, each of which contains two `.qmd` files, e.g., `01/01.qmd`, `01/01-a.qmd`, `02/02.qmd`, `02/02-a.qmd`, and so on.

When publishing a document (I only tried Netlify), I run the following command from the project root:

```
quarto publish netlify 01/01.qmd
```

This works, but somehow this single step adds temporary `*.quarto_ipynb` files in *all* subfolders (e.g., `01/01.quarto_ipynb`, `01/01-a.quarto_ipynb`, `02/02.quarto_ipynb`, `02/02-a.quarto_ipynb`, ...). Furthermore, these temporary files remain in the folders even when the publishing process has completed.

```
❯ quarto check
```

<details>

```
Quarto 1.6.36
[✓] Checking environment information...
      Quarto cache location: /Users/clemens/Library/Caches/quarto
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.4.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.46.3: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.6.36
      Path: /Users/clemens/Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/clemens/Library/TinyTeX/bin/universal-darwin
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.7
      Path: /Users/clemens/Projects/python-intro/.venv/bin/python
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.4.2
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library
      knitr: 1.49
      rmarkdown: 2.29

[✓] Checking Knitr engine render......OK
```
</details>

**Comment by mcanouil** [comment]
Do you observe the same behaviour using `quarto render`?

I'll move this as a discussion for now.

**Reply by cbrnr** [reply]
No, it only happens with `quarto publish`. Here's a screen capture of what's happening:


https://github.com/user-attachments/assets/f1c553cd-544e-4b14-a627-e2d0c505b1f0

The sources are available here in case they are needed: https://github.com/cbrnr/python-intro

**Reply by cscheid** [reply]
This looks like a bug to me. Do you think you could put together a minimal reprex? Thanks.

**Reply by cbrnr** [reply]
I tried replicating something with a similar structure from scratch, but I couldn't replicate the issue. So I think it's probably something very specific to this particular project. I'll try again and will report back, but maybe you can use the actual project I've linked meanwhile?

**Reply by mcanouil** [reply]
Do you have the issue with GitHub Page or Quarto Pub instead of Netlify?

**Reply by cscheid** [reply]
We can, but that lowers the fixing priority for us. I know you know the following, but to explain the math for everyone else:

Reducing test cases is something both you and we can do, but fixing the issue is something only we can do (and we have other things to do). So we can only work on reducing test cases when our plate is relatively empty, and that's uncommon.

**Reply by cbrnr** [reply]
Yes sure, I'll try as much as I can to help you. I've already confirmed that this issue does not occur with 1.5.57. I'll bisect a little to see when the bug was introduced.

**Reply by cbrnr** [reply]
I've narrowed it down to [v1.6.8](https://github.com/quarto-dev/quarto-cli/compare/v1.6.7...v1.6.8) (the last good release is v1.6.7). Maybe this already helps to identify the problem?

BTW, the issue also occurs for other providers (I tried publishing to `quarto-pub` and ended up with the same leftover files).

**Reply by mcanouil** [reply]
If you can reproduce with GitHub Pages as well, it would make it easier for us to debug already as no external tools would be needed.

**Reply by cbrnr** [reply]
@mcanouil I publish my website with GitHub Pages and I don't want to overwrite/modify anything. But with v1.6.8 I just did `quarto publish gh-pages 10/10-en.qmd` and the temp files appeared in all directories. Then I selected "No" (and the files were not deleted). So I guess I can reproduce with GitHub Pages, yes.

**Reply by cscheid** [reply]
> I've narrowed it down to [v1.6.8](https://github.com/quarto-dev/quarto-cli/compare/v1.6.7...v1.6.8) (the last good release is v1.6.7). Maybe this already helps to identify the problem?

Thanks. There isn't that much that can be blamed, but we did change the way in which we detect and resolve metadata from 1.6.7 to 1.6.8. It shouldn't have caused any externally-visible change... and yet, here we are.

**Reply by cbrnr** [reply]
Alright then, let me know if you want me to do anything else. BTW, and this is only slightly related, but was this an intentional decision to provide download links only for the latest RC at https://quarto.org/docs/get-started/? I download newer versions from time to time and wasn't even aware that this is not a stable release...

**Reply by mcanouil** [reply]
Yes, that's intentional. Note that it's written "Release Candidate".
<img width="484" alt="image" src="https://github.com/user-attachments/assets/e99a7d72-5415-4f63-828c-a5a191e3db81">

This happens a little bit before stable release to help discover possible major bugs.
It only switches the downloadable version on the Quarto site.
All other third party package manager still use the stable release, which can also be seen on GitHub as it's the latest from GitHub point of view:
<img width="216" alt="image" src="https://github.com/user-attachments/assets/8cc4b247-17ad-47aa-97ce-05d19e491e31">

**Reply by cbrnr** [reply]
I saw that Quarto 1.6 was released last week, is there anything I can do to help you fix this issue?

**Reply by mcanouil** [reply]
To fix which issue? Your remaining .quarto_ipynb?

Edit: the last week was mostly focussing on releasing 1.6.
We need to try ourselves what you reported, from there an issue will be opened.

**Reply by mcanouil** [reply]
Hum, I can reproduce using your repository @cbrnr 
The thing is that `quarto publish` on single file inside a project triggers the creation of `*.quarto_ipynb` for all documents present, but only the target gets cleaned up.

Here is a simple example where removing `_quarto.yml` solves the issue in this case.

```sh
touch _quarto.yml
for i in {1..10}; do
  echo -e "---\ntitle: File ${i}\nengine: jupyter\n---\n\n{{< lipsum 1 >}}" > file${i}.qmd;
done
git init
git remote add origin whatever
quarto publish gh-pages --no-browser file1.qmd
```

The publish command will crash because the remote is fake but you'll see that `.quarto_ipynb` are created for all instead of only for `file1.qmd`.

It seems like a bug but in a context of a project (default type) there is also some rational to the current behaviour.

**Reply by cbrnr** [reply]
The reason why I added an empty `_quarto.yml` file was that otherwise, I couldn't run commands from the root, e.g. `quarto publish 01/01.qmd`. I reported this in #5330, I'm not sure if this is still needed though.

Edit: I just tried with v1.6.39, and it seems if I delete my `_quarto.yml` then everything works! The problem only occurs when that file is present, whereas in previous versions it was the other way around (it *only* worked when this file existed).

**Reply by mcanouil** [reply]
Yes, it's because single file are now projects but I would still not expect to Quarto to create `.quarto_ipynb` files for all inputs in the project. `quarto render/preview` does not do that for instance.

**Reply by cbrnr** [reply]
OK, so you are saying this is still a bug then?

**Reply by mcanouil** [reply]
In my opinion but I am not the developer/designer of the initial behaviour and the change when single file became project, so I let mostly Carlos see what is what here.

**Reply by cscheid** [reply, marked as answer]
Yes, now that we have @mcanouil's repro I'm positive this is a [bug](https://github.com/quarto-dev/quarto-cli/issues/11597).
